v2/cmd/controller/app/setup.go

```diff
@@ -0,0 +1,352 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package app
+
+import (
+	"context"
+	"flag"
+	"math/rand"
+	"os"
+	"time"
+
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/benbjohnson/clock"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/Azure/azure-service-operator/v2/identity"
+	"github.com/Azure/azure-service-operator/v2/internal/config"
+	"github.com/Azure/azure-service-operator/v2/internal/controllers"
+	"github.com/Azure/azure-service-operator/v2/internal/crdmanagement"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+	asometrics "github.com/Azure/azure-service-operator/v2/internal/metrics"
+	armreconciler "github.com/Azure/azure-service-operator/v2/internal/reconcilers/arm"
+	"github.com/Azure/azure-service-operator/v2/internal/reconcilers/generic"
+	internalflags "github.com/Azure/azure-service-operator/v2/internal/util/flags"
+	"github.com/Azure/azure-service-operator/v2/internal/util/interval"
+	"github.com/Azure/azure-service-operator/v2/internal/util/kubeclient"
+	"github.com/Azure/azure-service-operator/v2/internal/util/lockedrand"
+	"github.com/Azure/azure-service-operator/v2/internal/version"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/conditions"
+)
+
+func SetupControllerManager(setupLog logr.Logger, flgs flags, ctx context.Context) manager.Manager {
+	scheme := controllers.CreateScheme()
+	_ = apiextensions.AddToScheme(scheme) // Used for managing CRDs
+
+	cfg, err := config.ReadAndValidate()
+	if err != nil {
+		setupLog.Error(err, "unable to get env configuration values")
+		os.Exit(1)
+	}
+
+	var cacheFunc cache.NewCacheFunc
+	if cfg.TargetNamespaces != nil && cfg.OperatorMode.IncludesWatchers() {
+		cacheFunc = cache.MultiNamespacedCacheBuilder(cfg.TargetNamespaces)
+	}
+
+	k8sConfig := ctrl.GetConfigOrDie()
+	mgr, err := ctrl.NewManager(k8sConfig, ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     flgs.metricsAddr,
+		NewCache:               cacheFunc,
+		LeaderElection:         flgs.enableLeaderElection,
+		LeaderElectionID:       "controllers-leader-election-azinfra-generated",
+		Port:                   9443,
+		HealthProbeBindAddress: flgs.healthAddr,
+	})
+
```

Comment:
minor: remove newline here as this is an error check for the above line and so I think makes sense to not have a newline between.